tests/functional/aws-node-sdk/test/bucket/getCorsResponse.js

```diff
@@ -0,0 +1,1083 @@
+import assert from 'assert';
+import http from 'http';
+import https from 'https';
+
+import { S3 } from 'aws-sdk';
+
+import conf from '../../../../../lib/Config';
+import getConfig from '../support/config';
+
+const config = getConfig('default', { signatureVersion: 'v4' });
+const s3 = new S3(config);
+
+const transport = conf.https ? https : http;
+const bucket = 'bucketcorstester';
+const hostname = process.env.AWS_ON_AIR ? 's3.amazonaws.com' :
+    'localhost';
+
+const port = process.env.AWS_ON_AIR ? 80 : 8000;
+
+const statusCode = {
+    200: 200,
+    NoSuchBucket: 404,
+    BadRequest: 400,
+    AccessForbidden: 403,
+};
+
+// AWS seems to take a bit long so sometimes by the time we send the request
+// the bucket has not yet been created or the bucket has been deleted.
+function _waitForAWS(callback, err) {
+    if (process.env.AWS_ON_AIR) {
+        setTimeout(() => callback(err), 5000);
+    } else {
+        callback(err);
+    }
+}
+
+function methodRequest(params, callback) {
+    const { method, objectKey, headers, code, headersResponse } = params;
+    const options = {
+        hostname,
+        port,
+        method,
+        headers,
+        path: objectKey ? `/${bucket}/${objectKey}` : `/${bucket}`,
+    };
+    const req = transport.request(options, res => {
+        const body = [];
+        res.on('data', chunk => {
+            body.push(chunk);
+        });
+        res.on('error', err => {
+            process.stdout.write('err receiving response');
+            return callback(err);
+        });
+        res.on('end', () => {
+            const total = body.join('');
+            if (code) {
+                const message = code === 200 ? '' : `<Code>${code}</Code>`;
+                assert(total.indexOf(message) > -1, `Expected ${message}`);
+                assert.deepEqual(res.statusCode, statusCode[code],
+                `status code expected: ${statusCode[code]}`);
+            }
+            if (headersResponse) {
+                Object.keys(headersResponse).forEach(key => {
+                    assert.deepEqual(res.headers[key], headersResponse[key],
+                      `error header: ${key}`);
+                });
+            } else {
```

Comment:
add comment: if no headersResponse provided, we should not have these headers in the request